src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java

```diff
@@ -161,64 +161,56 @@ public void complete(ValidationResult validationResult) {
                 TripPatternKey key = entry.getKey();
                 pattern.setStatementParameters(insertPatternStatement, true);
                 patternTracker.addBatch();
+                // Determine initial departure time based on the stop type.
+                int lastValidDepartureTime = isFlexStop(locationById, locationGroupById, key.stops.get(0))
+                    ? key.end_pickup_dropoff_window.get(0)
+                    : key.departureTimes.get(0);
```

Comment:
This block is repeated almost the same at line 323 in this file and in the unit test at line 21, in both cases to fill-in the `stopSequence === 0` case. Could that code be moved to `calculatePreviousDepartureTime`?